orangecontrib/educational/widgets/owgradientdescent.py

```diff
@@ -297,6 +339,51 @@ def __init__(self):
         self.step_size_lock()
         self.step_back_button_lock()
 
+    def _opt_changed(self):
+        self._show_right_optimizer()
+        self.select_optimizer()
+
+    def _show_right_optimizer(self):
+        enabled = [[False, False, False, False],  # SGD
+                   [True, False, False, False],  # Momentum
+                   [True, False, False, False],  # NAG
+                   [False, False, False, False],  # AdaGrad
+                   [False, True, False, False],  # RMSprop
+                   [False, True, False, False],  # AdaDelta
+                   [False, False, True, True],  # Adam
+                   [False, False, True, True],  # Adamax
+                ]
+
+        mask = [False, False, False, False]
+        self.combobox_opt.box.setVisible(self.stochastic)
+        if self.stochastic:
+            mask = enabled[self.opt_type]
+
+        # All hidden
+        for spin, enabled in zip(self._opt_params, mask):
+            [spin.box.hide, spin.box.show][enabled]()
+
+    def select_optimizer(self):
+        if self.learner is not None:
+            if self.opt_type == self._momentum:
+                self.learner.sgd_optimizer = Momentum(self.momentum)
+            elif self.opt_type == self._nag:
+                self.learner.sgd_optimizer = NesterovMomentum(self.momentum)
+            elif self.opt_type == self._adagrad:
+                self.learner.sgd_optimizer = AdaGrad()
+            elif self.opt_type == self._rmsprop:
+                self.learner.sgd_optimizer = RMSProp(self.rho)
+            elif self.opt_type == self._adadelta:
+                self.learner.sgd_optimizer = AdaDelta(self.rho)
+            elif self.opt_type == self._adam:
+                self.learner.sgd_optimizer = Adam(beta1=self.beta1,
+                                                  beta2=self.beta2)
+            elif self.opt_type == self._adamax:
+                self.learner.sgd_optimizer = Adamax(beta1=self.beta1,
+                                                    beta2=self.beta2)
+            else:
+                self.learner.sgd_optimizer = SGD()
```

Comment:
With all the more exotic parameters (beta1, beta2, ...) being tunable, why `learning_rate` isn't?